app/routes/stakeholder-router.js

```diff
@@ -1,15 +1,28 @@
 const router = require("express").Router();
 const stakeholderController = require("../controllers/stakeholder-controller");
+const jwtSession = require("../../middleware/jwt-session");
 
-router.get("/", stakeholderController.search);
-router.get("/dashboard", stakeholderController.searchDashboard);
-router.get("/:id", stakeholderController.getById);
-router.post("/csv", stakeholderController.csv);
-router.post("/", stakeholderController.post);
-router.put("/:id", stakeholderController.put);
-router.delete("/:id", stakeholderController.remove);
-router.put("/:id/needsVerification", stakeholderController.needsVerification);
-router.put("/:id/assign", stakeholderController.assign);
-router.put("/:id/claim", stakeholderController.claim);
+router.get("/", jwtSession.validateUser, stakeholderController.search);
+router.get(
+  "/dashboard",
+  jwtSession.validateUser,
+  stakeholderController.searchDashboard
+);
+router.get("/:id", jwtSession.validateUser, stakeholderController.getById);
+router.post("/csv", jwtSession.validateUser, stakeholderController.csv);
+router.post("/", jwtSession.validateUser, stakeholderController.post);
+router.put("/:id", jwtSession.validateUser, stakeholderController.put);
+router.delete("/:id", jwtSession.validateUser, stakeholderController.remove);
```

Comment:
for destructive calls like the `DELETE` we might want to limit it to certain access roles, like Admin or SuperAdmin. 
To do somethign like that we could extend the `jwtSession.validateUser` to take an option role e.g.

```javascript
router.delete(
  "/:id",
   jwtSession.validateUser({"allowRoles": ["admin", "superadmin"]}),
   stakeholderController.remove
);
```

Reply:
All of these should require an authenticated user.  The GET /dashboard, GET /:id and PUT /:id and POST /csv should require the dataEntry or admin role, the POST / and delete should require the admin role.

Reply:
Yup. It should be pretty easy to upgrade the validateUser middleware to be awaare of the roles.

Reply:
:+1: excellent ok I'll look for a relevant issue and if exists I'll assign to myself and work on it. If not i'll make one

Reply:
Issue #516 is all yours.  I'll work on #524 to get the client ready to handle the 401's by redirecting to a login page.